nidirect_common/src/InvalidateTaxonomyListCacheTags.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drupal\nidirect_common;
+
+use Drupal\Core\Cache\CacheTagsInvalidatorInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Class InvalidateTaxonomyListCacheTags.
+ */
+class InvalidateTaxonomyListCacheTags {
+
+  /**
+   * Constructs a new \Drupal\Core\Menu\MenuTreeStorage.
+   *
+   * @param \Drupal\Core\Cache\CacheTagsInvalidatorInterface $cache_tags_invalidator
+   *   The cache tags invalidator.
+   */
+  public function __construct(CacheTagsInvalidatorInterface $cache_tags_invalidator) {
+    $this->cacheTagsInvalidator = $cache_tags_invalidator;
```

Comment:
Argh, PHP! Does this not grumble about the class variable not existing before assignment in the constructor?

TL;DR: would expect to see a `protected $cacheTagsInvalidator;` class variable in there. If this works okay at runtime my guess would be that PHP assigns it on-the-fly which is both handy but also a bit frustrating for consistency :)

Reply:
Ah good spot, will fix it in the next PR as I am working in there again anyway